src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -446,20 +445,21 @@ private bool PartStartsWith(string candidate, TextSpan candidatePart, string pat
                 //      i.e. CoFiPro would match CodeFixProvider, but CofiPro would not.  
                 if (patternChunk.PatternHumps.Count > 0)
                 {
-                    var camelCaseKind = TryUpperCaseCamelCaseMatch(candidate, candidateHumps, patternChunk, CompareOptions.None, out var matchedSpans);
-                    if (camelCaseKind.HasValue)
+                    // No need to do the case sensitive check unless the case insensitive check is successful
```

Comment:
as per our offline convo, it might be nice to have a little more verbosity explaining why this is good.

namely that the expected case is to test the pattern matcher against thousands of items, with only a handful matching.  As such, the existing approach takes 2x checks against all those thousands of items.  Whereas the new approach does (1x + handful) checks.